src/control/server/mgmt_pool.go

```diff
@@ -167,73 +168,76 @@ func (svc *mgmtSvc) calculateCreateStorage(req *mgmtpb.PoolCreateReq) error {
 	if len(instances) < 1 {
 		return errors.New("harness has no managed instances")
 	}
-
 	if len(req.GetRanks()) == 0 {
 		return errors.New("zero ranks in calculateCreateStorage()")
 	}
+	if req.MetaBlobSize > 0 && !instances[0].GetStorage().BdevRoleMetaConfigured() {
+		return errors.New("meta size set in request but md-on-ssd is not enabled in config")
+	}
 
 	// NB: The following logic is based on the assumption that
 	// a request will always include SCM as tier 0. Currently,
 	// we only support one additional tier, NVMe, which is
 	// optional. As we add support for other tiers, this logic
 	// will need to be updated.
 
-	// the engine will accept only 2 tiers - add missing
-	if len(req.GetTierratio()) == 0 {
-		req.Tierratio = []float64{DefaultPoolScmRatio, DefaultPoolNvmeRatio}
-	} else if len(req.GetTierratio()) == 1 {
-		req.Tierratio = append(req.Tierratio, 0)
-	}
-
-	storagePerRank := func(total uint64) uint64 {
-		return total / uint64(len(req.GetRanks()))
-	}
+	nvmeMissing := !instances[0].GetStorage().HasBlockDevices()
 
-	if len(req.Tierbytes) == 0 {
-		req.Tierbytes = make([]uint64, len(req.Tierratio))
-	} else if len(req.Tierbytes) == 1 {
-		req.Tierbytes = append(req.Tierbytes, 0)
-	}
+	// As this is an exclusive interface between control-API and server, accept only known
+	// request parameter combinations.
 
 	switch {
-	case !instances[0].GetStorage().HasBlockDevices():
-		svc.log.Info("config has 0 bdevs; excluding NVMe from pool create request")
-		for tierIdx := range req.Tierbytes {
-			if tierIdx > 0 {
-				req.Tierbytes[tierIdx] = 0
-			} else if req.Tierbytes[0] == 0 {
-				req.Tierbytes[0] = storagePerRank(req.GetTotalbytes())
-			}
+	// Pool tier sizes already specified in request.
+	case len(req.Tierbytes) == 2 && len(req.Tierratio) == 0 && req.Totalbytes == 0:
+		// If no NVMe, refuse request as NVMe has been incorrectly requested.
+		nvmeBytes := req.Tierbytes[1]
+		if nvmeMissing && nvmeBytes > 0 {
+			return errors.Errorf("%s NVMe requested for pool but config has zero bdevs",
+				humanize.Bytes(nvmeBytes))
+		}
+
+	// Pool tier sizes to be populated based on total-size and ratio.
+	case len(req.Tierbytes) == 0 && len(req.Tierratio) == 2 && req.Totalbytes > 0:
+		// If no NVMe, adjust ratio as NVMe hasn't been specifically requested.
+		if nvmeMissing {
+			svc.log.Info("config has zero bdevs; excluding NVMe from pool create " +
```

Comment:
From my side, I would increase the log level to notice or warning as we change the input request.